src/transaction.ts

```diff
@@ -706,31 +717,57 @@ export class Snapshot extends EventEmitter {
       });
     };
 
-    return partialResultStream(this._wrapWithIdWaiter(makeRequest), {
-      json,
-      jsonOptions,
-      maxResumeRetries,
-      columnsMetadata,
-      gaxOptions,
-    })
-      ?.on('response', response => {
-        if (response.metadata && response.metadata!.transaction && !this.id) {
-          this._update(response.metadata!.transaction);
-        }
+    const q = {tableName: table, opts: this._observabilityOptions};
+    return startTrace('Snapshot.createReadStream', q, span => {
+      const prs = partialResultStream(this._wrapWithIdWaiter(makeRequest), {
+        json,
+        jsonOptions,
+        maxResumeRetries,
+        columnsMetadata,
+        gaxOptions,
       })
-      .on('error', err => {
-        const isServiceError = err && typeof err === 'object' && 'code' in err;
-        if (
-          !this.id &&
-          this._useInRunner &&
-          !(
-            isServiceError &&
-            (err as grpc.ServiceError).code === grpc.status.ABORTED
-          )
-        ) {
-          this.begin();
-        }
-      });
+        ?.on('response', response => {
+          if (response.metadata && response.metadata!.transaction && !this.id) {
+            this._update(response.metadata!.transaction);
+          }
+        })
+        .on('error', err => {
+          const isServiceError =
+            err && typeof err === 'object' && 'code' in err;
+          if (
+            !this.id &&
+            this._useInRunner &&
+            !(
+              isServiceError &&
+              (err as grpc.ServiceError).code === grpc.status.ABORTED
+            )
+          ) {
+            this.begin();
+          }
+          setSpanError(span, err);
+        })
+        .on('end', err => {
+          if (err) {
+            setSpanError(span, err);
+          }
+          span.end();
+        });
+
+      if (!prs) {
```

Comment:
I don't understand what is this check for ?

Reply:
There are instances where we have an EventEmitter returned that I found in tests, not being returned by the type  assertion to PartialResultStream.

Reply:
Can you please link the test, I am not able to get the reason for this check.

Reply:
@odeke-em Please see this

Reply:
This check was being defensive from some odd condition in earlier changes but now it doesn't fail. However, the others that check is instanceof Stream are valid.

Reply:
Removed, thank you @surbhigarg92!

Reply:
@surbhigarg92 actually I found where the problem was and it shows in the test
```shell
Transaction
       Snapshot
         createReadStream
           should use the transaction id if present:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
```
notice the 

```shell
TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
```

particularly for 
argument must be an instance of ReadableStream, WritableStream, or Stream. Received **undefined**